### PR TITLE
fix: ensure git user + email is setup

### PR DESCRIPTION
**Pull request by jstrachan**
lets refactor and add a reusable function too

### CODE REVIEW

File: pkg/jx/cmd/controller_build.go [hunk 1]
```diff
@@ -101,6 +101,12 @@ func (o *ControllerBuildOptions) Run() error {
 	o.EnvironmentCache = kube.CreateEnvironmentCache(jxClient, ns)
 
 	if o.InitGitCredentials {
+		// lets validate we have git configured
+		_, _, err = gits.EnsureUserAndEmailSetup(o.Git())
+		if err != nil {
+		  return err
```
Comment (ccojocar) [inline pkg/jx/cmd/controller_build.go]: I think would be nice to wrap this error with a message.

File: pkg/jx/cmd/controller_team.go [hunk 2]
```diff
@@ -92,28 +91,9 @@ func (o *ControllerTeamOptions) Run() error {
...
-		if userEmail == "" {
-			userEmail = "jenkins-x@googlegroups.com"
-		}
-		err = gitter.SetEmail("", userEmail)
-		if err != nil {
-			return errors.Wrapf(err, "Failed to set the git email to %s", userEmail)
-		}
+	_, _, err = gits.EnsureUserAndEmailSetup(co.Git())
+	if err != nil {
+	  return err
```
Comment (ccojocar) [inline pkg/jx/cmd/controller_team.go]: also here

File: pkg/jx/cmd/controller_build.go (hunk 1 above)
Comment (jstrachan) [reply pkg/jx/cmd/controller_build.go]: all the error conditions are already wrapped inside the `EnsureUserAndEmailSetup()` function so didn't see any need to wrap again

File: pkg/jx/cmd/controller_team.go (hunk 2 above)
Comment (jstrachan) [reply pkg/jx/cmd/controller_team.go]: ditto to reply in above conversation

File: pkg/gits/helpers.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"github.com/pkg/errors"
+	"os"
+	"os/user"
+)
+
+// EnsureUserAndEmailSetup returns the user name and email for the gitter
+// lazily setting them if they are blank either from the environment variables
+// `GIT_AUTHOR_NAME` and `GIT_AUTHOR_EMAIL` or using default values
+func EnsureUserAndEmailSetup(gitter Gitter) (string, string, error) {
```
Comment (agentgonzo) [inline pkg/gits/helpers.go]: would it make more sense to make this a receiver function on `Gitter`?

**Pull request merged**